pkgs/build-support/node/build-npm-package/hooks/npm-build-hook.sh

```diff
@@ -0,0 +1,17 @@
+npmBuildHook() {
+    echo "Executing npmBuildHook"
+
+    runHook preBuild
+
+    echo "Running build script"
+
+    echo "$PWD"
+
+    npm run "$npmBuildScript"
+
+    runHook postBuild
+}
+
+if [ -n "${npmBuildScript-}" ] && [ -z "${buildPhase-}" ]; then
```

Comment:
Why wouldn't `npmBuildScript` exist?

Reply:
Sometimes you want to add your own build phase -- this protects against not setting that while using the default build phase, at least I think that's the reason (stole from Cargo).

Reply:
Hmm, it sounds to me like that could cause a fair bit of confusion when including this hook manually. Sounds to me like we want to error out instead in that case, silently ignoring the build phase is not very intuative. 

Otherwise we end up not building anything which will cause the future phases to fail.